### DISCUSSION TITLE
WANTED: MacOS package maintainer

### DISCUSSION THREAD

**Discussion by robhagemans** [robhagemans/pcbasic#160] [Announcements]
Open call: if someone is willing to step up as Mac package maintainer and
knows how to create working app package releases for this platform, please contact me.

The existing cx_Freeze solution no longer seems to work due to Apple's active hostility to open source software (a.k.a. code signing requirements) and I no longer have the time or motivation to support this platform, so I will discontinue the Mac package starting with next release.

**Comment by incanus** [comment]
I'm game to try this (~20 years on Mac dev), though I have not yet gone through your packaging scripts to date. I too am frustrated by code signing, but I have the resources to muscle through it on the release schedule you've been operating. Not sure how best to coordinate, but feel free to reach out when you've tagged and I'll give it a shot.

**Comment by robhagemans** [comment]
Hey Justin, that's awesome!

I think the starting point would be the latest release 2.0.4 - there's a
dmg included that worked for me on an old MacOS install but apparently
fails on new ones due to code signing.

The packaging relies on cx-Freeze; the script packaging/mac.py build on
that. I'm not sure how familiar you are with cx-Freeze (me, not very, but
it was the best option to package Python for Windows and Mac ...) but it
has a somewhat unintuitive API that builds on setuptools.

It may just be a matter of running this on the latest MacOS, perhaps with a
newer version of cx-Freeze - unfortunately I'm not in a position to try
that as I don't have one :)

You'll also need the python packages listed in HACKING.md, in particular
SDL2, Pillow, lxml and markdown. Once you have a working recent Python
these can just be installed with pip (do new Macs still come with an
ancient Python version? It used to be a problem. I installed 3.7.9 from
python.org but you may be able to use a newer version). SDL2 needs to come
from libsdl.org but they have a Mac binary. SDL2-gfx is at ferzkopp.net and
I think needed compiling first - it's nice to have in the final package as
it allows for smooth scaling, but not essential to make it work on a first
try; it might be an additional headache you can do without...

Maybe you can have a look and see if any of it makes sense and if you can
make it run? Happy to answer questions/ talk through a bit more, I know my
existing packaging documentation is a bit sparse.
